source/Octopus.Cli/Util/CommandSuggester.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Octopus.Cli.Util
+{
+   public static class CommandSuggester
+    {
+        public static IEnumerable<string> SuggestCommandsFor(
+            string[] words,
+            IReadOnlyDictionary<string, string[]> completionItems)
+        {
+            words = words.Except(new[] { "octo", "complete" }).ToArray();
+            var numberOfArgs = words.Length;
+            var searchTerm = numberOfArgs > 0 ? words.Last() ?? "" : "";
+            var suggestions = new List<string>();
+            var isOptionSearch = searchTerm.StartsWith("--");
```

Comment:
what about abbreviated args?

Reply:
Thinking that might be a later thing. AFAIK the only ones here are `help` and `version`. I felt that the main point of this feature is discoverability, and being able to resolve `-h` or `-ver` didn't seem worth the effort.

Reply:
(as opposed to resolving `--help` or `--version`)